### PR TITLE
Add traceback to the error dialog

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: simphony_ui/ui.py [hunk 1]
```diff
@@ -110,8 +111,16 @@ class Application(HasStrictTraits):
     )
 
     @on_trait_change('calculation_error_event', dispatch='ui')
-    def show_error(self, msg):
-        error(None, 'Oups ! Something went bad:\n{}'.format(msg), 'Error')
+    def show_error(self, messages):
+        error_msg, tb_msg = messages
+        error(
+            None,
+            'Oups ! Something went bad:\n{}\n\nDetails:\n{}'.format(
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: You can pass the detail as a special argument. The dialog will show a "more..." button
```
 71 def error(parent, message, title='Error', detail='', informative=''):              
```

File: simphony_ui/ui.py [hunk 2]
```diff
@@ -218,7 +227,8 @@ def _run_calc_threaded(self):
                 self.update_progress_bar
             )
         except Exception as e:
-            self.calculation_error_event = str(e)
+            tb = traceback.format_exc().splitlines()
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: There's no need to split it and then join it again with '\n' just pass traceback.format_exc() to the tuple.

**Review by stefanoborini — CHANGES_REQUESTED** [review]

File: simphony_ui/ui.py (hunk 2 above)
Comment (martinRenou) [reply simphony_ui/ui.py]: Oh yes that's stupid. I first wanted to remove the last line, that's why I wanted to split the message.

File: simphony_ui/ui.py (hunk 1 above)
Comment (martinRenou) [reply simphony_ui/ui.py]: Hum... Are we really talking about the pyface.api.error function ? Because I don't have detail and informative arguments.

Comment (martinRenou) [reply simphony_ui/ui.py]: Ok it sounds like my version of pyface is an old version

**Review by stefanoborini — APPROVED** [review]

**Pull request merged**